src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1094,21 +1084,36 @@ private void NotifyDatabaseAboutChanged(TransactionOperationContext context, str
             context.Transaction.InnerTransaction.LowLevelTransaction.OnDispose += transaction =>
             {
                 if (transaction is LowLevelTransaction llt && llt.Committed)
-                    TaskExecutor.Execute(_ =>
-                    {
-                        try
-                        {
-                            DatabaseChanged?.Invoke(this, (databaseName, index, type, change));
-                            _rachisLogIndexNotifications.NotifyListenersAbout(index, null);
-                        }
-                        catch (Exception e)
-                        {
-                            _rachisLogIndexNotifications.NotifyListenersAbout(index, e);
-                        }
-                    }, null);
+                {
+                    ExecuteAsyncTask(index, () => DatabaseChanged?.Invoke(this, (databaseName, index, type, change)));
+                }
             };
         }
 
+        private void ExecuteAsyncTask(long index, Action action)
+        {
+            _rachisLogIndexNotifications.AddTask(index);
```

Comment:
A comment is needed here to explain why we do this _before_ starting the async task